Sources/ToolchainRegistry/Toolchain.swift

```diff
@@ -366,10 +366,20 @@ public final class Toolchain: Sendable {
 }
 
 /// Find a containing xctoolchain with plist, if available.
-func containingXCToolchain(
+private func containingXCToolchain(
   _ path: URL
 ) -> (XCToolchainPlist, URL)? {
-  var path = path
+  // `deletingLastPathComponent` only makes sense on resolved paths (ie. those without symlinks or `..`). Any given
+  // toolchain path should have already been realpathed, but since this can turn into an infinite loop otherwise, it's
+  // better to be safe than sorry.
+  guard let resolvedPath = try? path.realpath else {
```

Comment:
Use `orLog` here so that we get some indication in case `realpath` fails, in which case we would only return `nil` here without logging the fault below?

Reply:
Sure, I didn't want to be too spammy here but... 🤷